automation/build-bin.ts

```diff
@@ -354,6 +370,41 @@ export async function buildOclifInstaller() {
 	let packOpts = ['-r', ROOT];
 	if (process.platform === 'darwin') {
 		packOS = 'macos';
+		await new Promise((resolve, reject) => {
+			klaw('node_modules/')
+				.on('data', (item: { path: string; stats: Stats }) => {
+					if (!item.stats.isFile()) {
+						return;
+					}
+					if (
+						path.basename(item.path).endsWith('.zip') &&
+						path.dirname(item.path).includes('test')
+					) {
+						console.log('Removing zip', item.path);
+						fs.unlinkSync(item.path);
+					}
+				})
+				.on('end', resolve)
+				.on('error', reject);
+		});
+		// Sign all .node files first
+		await new Promise((resolve, reject) => {
+			klaw('node_modules/')
+				.on('data', async (item: { path: string; stats: Stats }) => {
+					if (!item.stats.isFile()) {
+						return;
+					}
+					if (path.basename(item.path).endsWith('.node')) {
+						await whichSpawn('productsign', [
```

Comment:
Oh, can't use `await` inside an event handler like `on('data')`. The result is `UnhandledPromiseRejectionWarning` if the promise is rejected - https://ci.balena-dev.com/builds/803380
```
Usage: productsign [options] --sign <identity> <input-product-path> <output-product-path>

(node:65442) UnhandledPromiseRejectionWarning: Error: productsign failed with exit code 1:
"/usr/bin/productsign" [--sign,Developer ID Installer: Rulemotion Ltd (66H43P8FRG),/private/var/lib/concourse/worker/volumes/live/797930ff-065b-4eac-71a4-afa1b720e87e/volume/node_modules/fsevents/fsevents.node]
    at Object.whichSpawn (/private/var/lib/concourse/worker/volumes/live/797930ff-065b-4eac-71a4-afa1b720e87e/volume/automation/utils.ts:236:9)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:65442) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:65442) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
So there are 2 separate errors there:
* `productsign` exits with error code 1 and the only debugging output for that is:  
`Usage: productsign [options] --sign <identity> <input-product-path> <output-product-path>`  
I.e., a malformed command line. The way to further debug this is to run `productsign` manually on a fly shell, and/or to add a `console.error(...)` message printing the full command line before attempting to execute it.

* `UnhandledPromiseRejectionWarning` - this is thrown because the promise rejection is not handled in the `on('data')` event handler. The promise is rejected because `productsign` returns a non-zero error code.